deps/wazuh_testing/wazuh_testing/tools/remoted_sim.py

```diff
@@ -364,36 +416,40 @@ def start_connection(self):
                         time.sleep(1)
                         if connection:
                             connection.close()
-                        
+
                 return connection, client_address
             except Exception as identifier:
                 continue
-     
+
     """
     Listener thread that will finish when encryption_keys are obtained
     """
-    def upgrade_listener(self, filename, filepath, chunk_size, installer, 
+    def upgrade_listener(self, filename, filepath, chunk_size, installer,
                          sha1hash, simulate_interruption=False,
-                         simulate_connection_error=False):   
+                         simulate_connection_error=False):
         self.upgrade_errors = False
         self.upgrade_success = False
-        while not self.upgrade_errors and self.running: 
+
+        while not self.upgrade_errors and self.running:
             try:
                 connection, client_address = self.start_connection()
                 self.sendComMessage(client_address, connection, 'lock_restart -1')
-                self.sendComMessage(client_address, connection, f'open wb {filename}', 
+                self.sendComMessage(client_address, connection, f'open wb {filename}',
                                     interruption_time=5 if simulate_interruption else None)
                 with open(filepath, 'rb') as f:
                     bytes_stream = f.read(chunk_size)
+                    bytes_stream_64 = base64.b64encode(bytes_stream)
```

Comment:
This is only necessary for agents that have a version installed <= 4.1.

Agents with version > 4.1 still process the commands with execd, which expects the chunk in binary and not in base64.